tensorflow/cc/gradients/nn_grad.cc

```diff
@@ -47,6 +48,38 @@ Status SoftmaxGrad(const Scope& scope, const Operation& op,
 }
 REGISTER_GRADIENT_OP("Softmax", SoftmaxGrad);
 
+Status SoftmaxCrossEntropyWithLogitsGrad(const Scope& scope,
+                                          const Operation& op,
+                                          const std::vector<Output>&
+                                          grad_inputs,
+                                          std::vector<Output>* grad_outputs) {
+  // Softmax gradient with cross entropy logits function
+  // We multiply the backprop for cost with the gradients - op.output[1]
+  // There is no gradient for labels
+
+  auto softmaxGrad = op.output(1);
+  auto gradLoss = grad_inputs[0];
+  auto gradGrad = grad_inputs[1];
+
+  auto tempGrad = Mul(scope, gradLoss, softmaxGrad);
+
+  // TODO: Check if this sufficient, need a check for null ?
+  if (gradGrad.op().output_type(0) != 0) {
```

Comment:
Python defines a local function:
```python
 def IsZero(g):
    # Some introspection to check if the gradient is feeding zeros
    if g.op.type in ("ZerosLike", "Zeros"):
      return True
    const_fill_value = tensor_util.constant_value(g)
    return const_fill_value is not None and (const_fill_value == 0).all()
```
probably not the same, check probably needs to be extended to cover same cases